src/components/NewItemForm.jsx

```diff
@@ -55,14 +103,111 @@ const NewItemForm = () => {
   const [endpoint, setEndpoint] = useState("");
   const [password, setPassword] = useState("");
   const [username, setUserName] = useState("");
+  const [itemNameError, setItemNameError] = useState(clearError);
+  const [endpointError, setEndpointError] = useState(clearError);
+  const [passwordError, setPasswordError] = useState(clearError);
+  const [usernameError, setUserNameError] = useState(clearError);
+
   const [collectionID, setCollectionID] = useState("");
+  const [collectionArray, setCollectionArray] = useState([]);
+  const [defaultCollectionsID, setDefaultCollectionID] = useState("");
+
+  const [symmetricKey, setSymmetricKey] = useState("");
 
   const handleCollectionChange = (event) => {
     setCollectionID(event.target.value);
   };
 
-  const onSaveVaultItem = () => {
-    console.log("status");
+  const getUserItems = async () => {
+    try {
+      let collectionResponse = await getCollections(getCollections);
+      let keyResponse = await getUserKey();
+      setCollectionArray(collectionResponse.data);
+
+      // We need to decrypt this in the future
+      let encryptedSymmetricKey = keyResponse.data.encrypted_symmetric_key;
```

Comment:
Unless I'm missing something we don't encrypt the vault collection  names using they key so I don't think we need it.

Reply:
The call here is to get the key to be used later with vault items and it won't be used on collections. I decided to combine both the collections and the get user key in one call, but I can see now that the naming can lead to confusion. I will rename the function from getUserItems to maybe getUserDetails